### ISSUE TITLE
VncViewer stuck in fl::wait()

### ISSUE THREAD

**Issue by xekkoman** [TigerVNC/tigervnc#922]
Good day. 
I encountered bug using vncviewer under centos 7.7.1908.
vncviewer version is 1.10.0-4.el7.x86_64

Test case is:
1) Start script
while true 
do 
	vncviewer -passwd 7.vnc -AlertOnFatalError=0 -ViewOnly *WINDOWS PC ADDRES"
	sleep 1
done

2) Connect to windows machine with vncserver
3) Reboot windows machine
4) VncViewer will restart and reconnect, you will see windows shutting down screen.
5) After that vnc will wait forever for mouse or keyboard event, and don`t disconnect from server that closed connection.

gdb backtrace:
(gdb) bt
#0  0x00007f6eabd3cbd3 in select () from /usr/lib64/libc.so.6
#1  0x00007f6eaea86a56 in fl_wait(double) () from /usr/lib64/libfltk.so.1.3
#2  0x00007f6eaea2786f in Fl::wait(double) () from /usr/lib64/libfltk.so.1.3
#3  0x000000000042f056 in run_mainloop() ()
#4  0x000000000041fc25 in main ()

I think, main problem is that network events processed only after keyboard or mouse events acquired. 
Because without -ViewOnly flag, vncviewer works properly.

**Comment by CendioOssman** [comment]
It sounds like the Windows machine is rebooting without fully closing all TCP connections. At that point the viewer will most likely get stuck as it will not detect that the network is down until it tries to send something (which it won't do when in ViewOnly mode).

I'm afraid I don't have an easy workaround for this.

**Comment by xekkoman** [comment]
Created pull request, its not directly solves that problem, but still can be useful:
Screen refresh timer. #941
